docs/topics/data-disks.md

```diff
@@ -0,0 +1,88 @@
+# Data Disks
+
+This document describes how to specify data disks to be provisioned and attached to VMs provisioned in Azure. 
+
+## Azure Machine Data Disks
+
+Azure Machines support optionally specifying a list of data disks to be attached to the virtual machine. Each data disk must have:
+ - `nameSuffix` - the name suffix of the disk to be created. Each disk will be named `<machineName>_<nameSuffix>` to ensure uniqueness. 
```

Comment:
Do the names need to be unique for the disks? If so, then it might be good to add that to the validation

Reply:
they do, will add to validation